avm-transpiler/src/utils.rs

```diff
@@ -0,0 +1,41 @@
+use log::debug;
+
+use acvm::acir::circuit::brillig::Brillig;
+use acvm::acir::circuit::Opcode;
+
+use crate::instructions::AvmInstruction;
+
+/// Extract the Brillig program from its ACIR wrapper instruction.
+/// An Noir unconstrained function compiles to one ACIR instruction
+/// wrapping a Brillig program. This function just extracts that Brillig
+/// assuming the 0th ACIR opcode is the wrapper.
+pub fn extract_brillig_from_acir(opcodes: &Vec<Opcode>) -> &Brillig {
+    if opcodes.len() != 1 {
+        panic!("There should only be one brillig opcode");
+    }
+    let opcode = &opcodes[0];
+    let brillig = match opcode {
+        Opcode::Brillig(brillig) => brillig,
+        _ => panic!("Tried to extract a Brillig program from its ACIR wrapper opcode, but the opcode doesn't contain Brillig!"),
+    };
+    brillig
+}
+
+/// Print inputs, outputs, and instructions in a Brillig program
+pub fn dbg_print_brillig_program(brillig: &Brillig) {
```

Comment:
these can be implemented thorugh the debug trait. https://doc.rust-lang.org/std/fmt/trait.Debug.html

Reply:
That would require adding the trait in the Noir subrepo, correct? I'm happy to submit a separate PR to move this there.

Reply:
Hmm maybe, not a blocker